### PR TITLE
Updated security-related documetation

### PR DESCRIPTION
**Pull request by zbeech**
- Added key-store docs
- Added known-hosts-file docs
- Updated VIP authentication and authorization docs
- Updated protecting pub/sub docs
- Added road-map with links to security-related docs

---
This change is [Reviewable](https://reviewable.io/reviews/volttron/volttron/919)

### CODE REVIEW

File: docs/source/core_services/messagebus/VIP/VIP-Authentication.rst [hunk 1]
```diff
@@ -24,7 +25,7 @@ Internet. Therefore, VOLTTRON automatically generates an encryption key
 and enables `CurveMQ <http://rfc.zeromq.org/spec:26>`__ by default on
 all TCP connections.
 
-The key is stored in ``$VOLTTRON_HOME/curve.key``, where VOLTTRON\_HOME
+The key is stored in ``$VOLTTRON_HOME/keystore.json``, where VOLTTRON\_HOME
 defaults to ``$HOME/.volttron``. The base64-encode public key is printed
 in the log output at the **INFO** level when VOLTTRON starts (with -v
```
Comment (craig8) [inline docs/source/core_services/messagebus/VIP/VIP-Authentication.rst]: Why aren't we having them use volltron-ctl auth serverkey?

File: docs/source/core_services/security/Key-Stores.rst [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+These keys are saved in a key store, which is implemented by the
+:doc:`KeyStore class<volttron.platform.keystore.KeyStore>`.
+Each agent has its own key store.
+
+Key Store Locations
+-------------------
+
+There are two main locations key stores will be saved. Installed agents'
+key stores are in the the agent's data directory::
+
```
Comment (craig8) [inline docs/source/core_services/security/Key-Stores.rst]: Don't placeholders in man pages use less than greater than signs around the placeholder?

File: docs/source/core_services/security/Known-Hosts-File.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+~~~~~~~~~~~~~
+
+Here are the contents of an example known-hosts file:
+
+.. code:: JSON
+
+    {
+        "@": "FSG7LHhy3v8tdNz3gK35G6-oxUcyln54pYRKu5fBJzU", 
+        "127.0.0.1:22916": "FSG7LHhy3v8tdNz3gK35G6-oxUcyln54pYRKu5fBJzU", 
+        "192.168.0.42:22916": "uhjbCUm3kT5QWj5Py9w0XZ7c1p6EP8pdo4Hq4dNEIiQ"
```
Comment (craig8) [inline docs/source/core_services/security/Known-Hosts-File.rst]: Does it add one for each bound address?  I can have multiple VIP addresses on an instance.

File: docs/source/core_services/messagebus/VIP/VIP-Authentication.rst (hunk 1 above)
Comment (zbeech) [inline docs/source/core_services/messagebus/VIP/VIP-Authentication.rst]: Good point. I'll change that.

File: docs/source/core_services/security/Key-Stores.rst (hunk 2 above)
Comment (zbeech) [inline docs/source/core_services/security/Key-Stores.rst]: Yes, that's more common. I'll change that.

File: docs/source/core_services/security/Known-Hosts-File.rst (hunk 3 above)
Comment (zbeech) [inline docs/source/core_services/security/Known-Hosts-File.rst]: Yes, it adds one for each address. I'll mention that in the docs.

**Comment by zbeech** [comment]
Review status: 0 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

*[docs/source/core_services/messagebus/VIP/VIP-Authentication.rst, line 30 at r3](https://reviewable.io:443/reviews/volttron/volttron/919#-KUUGr4AdX3k_JldsXqh-r3-30:-KUXuUrAkzD5IgPNFefp:b-896fix) ([raw file](https://github.com/volttron/volttron/blob/6b968a21c921be9167c508ef81bc96d2882f8505/docs/source/core_services/messagebus/VIP/VIP-Authentication.rst#L30)):*
<details><summary><i>Previously, zbeech wrote…</i></summary>
> Good point. I'll change that.
</details>
Done.

---

*[docs/source/core_services/security/Key-Stores.rst, line 23 at r3](https://reviewable.io:443/reviews/volttron/volttron/919#-KUUGr4AdX3k_JldsXql-r3-23:-KUXuWkOpTxoJmOCLoEn:b-896fix) ([raw file](https://github.com/volttron/volttron/blob/6b968a21c921be9167c508ef81bc96d2882f8505/docs/source/core_services/security/Key-Stores.rst#L23)):*
<details><summary><i>Previously, zbeech wrote…</i></summary>
> Yes, that's more common. I'll change that.
</details>
Done.

---

*[docs/source/core_services/security/Known-Hosts-File.rst, line 58 at r3](https://reviewable.io:443/reviews/volttron/volttron/919#-KUUGr4AdX3k_JldsXqm-r3-58:-KUXuXKF_JTmIHeNq7NK:b-896fix) ([raw file](https://github.com/volttron/volttron/blob/6b968a21c921be9167c508ef81bc96d2882f8505/docs/source/core_services/security/Known-Hosts-File.rst#L58)):*
<details><summary><i>Previously, zbeech wrote…</i></summary>
> Yes, it adds one for each address. I'll mention that in the docs.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/volttron/volttron/919)*

**Comment by zbeech** [comment]
@craig8 Are there any unresolved issues with this pull request?

**Pull request merged**